lib/matplotlib/tests/test_cycles.py

```diff
@@ -104,6 +126,45 @@ def test_fillcycle_ignore():
     ax.legend(loc='upper left')
 
 
+@cleanup
+def test_valid_input_forms():
+    fig, ax = plt.subplots()
+    # These should not raise an error.
+    ax.set_prop_cycle(None)
+    ax.set_prop_cycle(cycler('linewidth', [1, 2]))
+    ax.set_prop_cycle('color', 'rgywkbcm')
+    ax.set_prop_cycle('linewidth', (1, 2))
+    ax.set_prop_cycle('linewidth', [1, 2])
+    ax.set_prop_cycle('linewidth', iter([1, 2]))
+    ax.set_prop_cycle('linewidth', np.array([1, 2]))
+    ax.set_prop_cycle('color', np.array([[1, 0, 0],
+                                         [0, 1, 0],
+                                         [0, 0, 1]]))
+    ax.set_prop_cycle(lw=[1, 2], color=['k', 'w'], ls=['-', '--'])
+    ax.set_prop_cycle(lw=np.array([1, 2]),
+                      color=np.array(['k', 'w']),
+                      ls=np.array(['-', '--']))
+    assert True
+
+
+@cleanup
+def test_invalid_input_forms():
+    fig, ax = plt.subplots()
+    with assert_raises((TypeError, ValueError)):
```

Comment:
I knew this was going to be confusing when we decided this awhile ago...

It is that official support for 2.6 ended for v1.5, but it seems that we still have Travis testing v1.5.x branch with python2.6. So, whoever backports this PR should be aware of that.